src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -256,43 +256,57 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
       this.search = '';
     }
     const columns = this.columns.filter(c => c.cellTransformation !== CellTemplate.sparkline);
+    if (this.search.match(/['"][^'"]+['"]/)) {
+      this.search = this.search.replace(/['"][^'"]+['"]/g, (match: string) => {
+        return match.replace(/(['"])([^'"]+)(['"])/, '$2').replace(' ', '+');
+      });
+    }
     // update the rows
-    this.rows = this.subSearch(this.data, this.search.toLowerCase().split(/[, ]/), columns);
+    this.rows = this.subSearch(this.data, this.search.toLowerCase().split(/[ ]/), columns);
     // Whenever the filter changes, always go back to the first page
     this.table.offset = 0;
   }
 
   subSearch (data: any[], currentSearch: string[], columns: CdTableColumn[]) {
-    let tempColumns: CdTableColumn[];
     if (currentSearch.length === 0 || data.length === 0) {
       return data;
     }
-    const searchWords: string[] = currentSearch.pop().split(':');
-    if (searchWords.length === 2) {
-      tempColumns = [...columns];
-      columns = columns.filter((c) => c.name.toLowerCase().indexOf(searchWords[0]) !== -1);
+    const searchTerms: string[] = currentSearch.pop().split(':');
+    const columnsClone = [...columns];
+    const dataClone = [...data];
+    const getSearchTerm = (s: string) => s.replace('+', ' ');
+    const filterColumns = (columnName: string) =>
+      columnsClone.filter((c) => c.name.toLowerCase().indexOf(columnName) !== -1);
+    if (searchTerms.length === 2) {
+      columns = filterColumns(getSearchTerm(searchTerms[0]));
     }
-    const searchWord: string = _.last(searchWords);
-    if (searchWord.length > 0) {
-      data = data.filter(d => {
-        return columns.filter(c => {
-          let cellValue: any = _.get(d, c.prop);
-          if (_.isUndefined(cellValue)) {
-            return;
-          }
-          if (_.isArray(cellValue)) {
-            cellValue = cellValue.join('');
-          } else if (_.isNumber(cellValue)) {
-            cellValue = cellValue.toString();
-          }
-          return cellValue.toLowerCase().indexOf(searchWord) !== -1;
-        }).length > 0;
-      });
+    const searchTerm: string = getSearchTerm(_.last(searchTerms));
+    data = this.basicDataSearch(searchTerm, data, columns);
+    // Checks if user searches for column but he is still typing
```

Comment:
I think the search should focus only on the values of each field, it should only check the column if there is a `:` after the string.

You could delay the search until the user finished typing, that way you would never get a empty result caused by an incomplete search.

Reply:
This way you also won't get an empty list, and it's a nicer user experience to provide instant search results.